test/provisioner/kustomize.go

```diff
@@ -78,3 +85,121 @@ func (kh *KustomizeOverlay) YamlReload() error {
 
 	return nil
 }
+
+// SetKustomizeConfigMapGeneratorLiteral updates the kustomization YAML by setting `value` to `key` on the
+// `cmgName` ConfigMapGenerator literals. If `key` does not exist then a new entry is added.
+func (kh *KustomizeOverlay) SetKustomizeConfigMapGeneratorLiteral(cmgName string, key string, value string) (err error) {
```

Comment:
I wonder if the cmgName should be fixed/const, it seems to be added by the implementation of Edit, however, the naming of the CM must match the implementation and also for the peerpodsconfig controller ( it is the same across all providers)

Reply:
Hi @snir911 , while the name is currently the same across providers I prefer to keep this function as generic as possible so that it can be used in other contexts if needed.